packages/ckeditor5-dev-env/tests/translations/upload.js

```diff
@@ -10,67 +10,82 @@
 const path = require( 'path' );
 const sinon = require( 'sinon' );
 const mockery = require( 'mockery' );
-const transifexService = require( '../../lib/translations/transifex-service' );
-const fs = require( 'fs' );
 
 describe( 'upload', () => {
-	let sandbox;
+	const sandbox = sinon.sandbox.create();
 	let upload;
+	let packageNames;
+	let stubs;
 
 	beforeEach( () => {
-		sandbox = sinon.sandbox.create();
+		mockery.enable( {
+			useCleanCache: true,
+			warnOnReplace: false,
+			warnOnUnregistered: false,
+		} );
+
+		stubs = {
+			logger: {
+				info: sandbox.stub(),
+				warning: sandbox.stub(),
+				error: err => {
+					throw err;
+				},
+			},
+
+			transifexService: {
+				getResources: sandbox.spy( () => Promise.resolve( [ {
+					slug: 'ckeditor5-core'
+				} ] ) ),
+				postResource: sandbox.spy( () => Promise.resolve( '[]' ) ),
+				putResourceContent: sandbox.spy( () => Promise.resolve( '{}' ) ),
+			},
+
+			fs: {
+				readdirSync: sandbox.spy( () => packageNames ),
+				createReadStream: sandbox.spy( path => `${ path } content` ),
```

Comment:
You can try to use a mock which I wrote for `conventionalChangelog()` function in `generateChangelog()` task - https://github.com/ckeditor/ckeditor5-dev/blob/master/packages/ckeditor5-dev-env/tests/release-tools/tasks/generatechangelog.js#L26-L34.